### PR TITLE
Adding base NU5 declarations and logic

### PR DESCRIPTION
**Pull request by steven-ecc**
Add NU5 base definitions and boilerplate logic.

Closes [#5025](https://github.com/zcash/zcash/issues/5025).

### CODE REVIEW

File: src/consensus/upgrades.cpp [hunk 1]
```diff
@@ -43,6 +43,11 @@ const struct NUInfo NetworkUpgradeInfo[Consensus::MAX_NETWORK_UPGRADES] = {
         /*.nBranchId =*/ 0xe9ff75a6,
         /*.strName =*/ "Canopy",
         /*.strInfo =*/ "See https://z.cash/upgrade/canopy/ for details.",
+    },
+    {
+        /*.nBranchId =*/ 0xf919a198,
+        /*.strName =*/ "NU5",
+        /*.strInfo =*/ "See https://z.cash/upgrade/nu5/ for details.",
```
Comment (nuttycom) [inline src/consensus/upgrades.cpp:50]: This page will need to be created.

**Review by nuttycom — APPROVED** [review]
Looks correct!

**Review by daira — APPROVED** [review]
utACK

**Comment by steven-ecc** [comment]
@zkbot try

**Comment by daira** [comment]
@zkbot try

**Comment by steven-ecc** [comment]
Looks like a transient CI failure.  @zkbot try

**Comment by steven-ecc** [comment]
@zkbot r+

**Pull request merged**